### ISSUE TITLE
FileInfo_Query_FileBasicInfo_File_IsIntegritySupported always fails if OTHERFS and named streams

### ISSUE THREAD

**Issue by gwr** [microsoft/WindowsProtocolTestSuites#260]
In the test log, I see this failure:

```
2022-01-14 20:22:38.409 [TestStep] 1. Create DataFile with FileAttribute.INTEGRITY_STREAM&#xD;
2022-01-14 20:22:38.414 [Checkpoint] Section 2.1.5.1
                                                                Phase 6 -- Location of file:
                                                                If StreamTypeNameToOpen is non-empty and StreamTypeNameToOpen is not equal to one of the stream type names
                                                                recognized by the object store&lt;42&gt; (using case-insensitive string comparisons), the operation MUST be failed
                                                                with STATUS_OBJECT_NAME_INVALID.&#xD;
2022-01-14 20:22:38.414 [CheckFailed] Assert.AreEqual failed. Expected: &lt;OBJECT_NAME_INVALID (Microsoft.Protocols.TestSuites.FileSharing.FSA.Adapter.MessageStatus)&gt;, Actual: &lt;SUCCESS (Microsoft.Protocols.TestSuites.FileSharing.FSA.Adapter.MessageStatus)&gt;. return of CreateFile&#xD;
```

This test has a large section of code starting here:
  TestSuites/FileServer/src/FSA/Adapter/FSAAdapter.cs : Line 684

```
            /*
             * Work around for test cases only designed for NTFS and ReFS:
             * Make assertion in the adapter, then convert the return code according to the test case.
             */
            if (this.fileSystem != Adapter.FileSystem.NTFS && this.fileSystem != Adapter.FileSystem.REFS)
            {
               // ... questions about this ...
            }
```

It looks like this code is trying to verify that invalid stream types result in STATUS_OBJECT_NAME_INVALID
but the stream name being given to our server is actually "something::$DATA" which is in fact valid.
I'm not sure I understand the lists of stream types shown in that big "if" section, but something is
apparently not working as expected when the FS is "OTHERFS" and named streams are supported.

BTW, in our case OTHERFS does not pretend to offer any of the "Is integrity supported" features.
ZFS does technically have "integrity" features, but we do not try to support any of the FSA interfaces
for checking or manipulating integrity features as might be allowed on REFS.

Suggestions appreciated.
Thanks!

**Comment by gwr** [comment]
One general suggestion for this test case might be to introduce a new parameter to tell the test what string to use as an invalid stream type (instead of trying to guess all the valid stream types).

**Comment by Dingshouqing** [comment]
Hi Gordon,
 
The code you mentioned is aligned with [MS-FSA  Appendix <45> and <46>](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fsa/4e3695bd-7574-4f24-a223-b4679c065b63#Appendix_A_45) only for windows file systems. 
We’ll update our code to add check OS platform 
 ![2](https://user-images.githubusercontent.com/17266569/150105634-2c9474ca-3b72-495f-97c0-c91f42154663.png)
As all other windows file systems (except NTFS and REFS) do not recognize any stream type names. we’ll check if the file contains a complex name suffix instead of checking if it contains valid or invalid stream types.
For non-windows platforms, we will not check the complex name suffixes.
 
We will keep you updated once it is done.

Best Regards,
Windows Protocol Test Suites Development Team

**Comment by Dingshouqing** [comment]
Hi Gordon,

This issue has been fixed and the code has already been merged into the main branch by commit [57a06cd ](https://github.com/microsoft/WindowsProtocolTestSuites/commit/57a06cd1561fec83bf28637ee03c295daa830ba7), feel free to clone the latest code and build to have a try.

After merging the fix, if your File system support alternate data stream please update the below property and add OTHERFS into the value
```
<Property name="WhichFileSystemSupport_AlternateDataStream" value="NTFS;REFS;OTHERFS" />
```
Your file system will check if it supports the ADS attribute instead of our test code

Best Regards,
Windows Protocol Test Suites Development Team

**Comment by Dingshouqing** [comment]
Hi Gordon,

Since this thread has been inactive for some time, we will close it for now. Please feel free to contact us if there is any further question.

Best Regards,
Windows Protocol Test Suites Development Team
